src/veritysetup/veritysetup-generator.c

```diff
@@ -191,47 +246,62 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
         return 0;
 }
 
-static int determine_devices(void) {
+static int determine_device(char *hash, char **data_what_p, char **hash_what_p) {
```

Comment:
Constify `hash` and drop `_p` suffix:
`determine_device(const char *hash, char **data_what, char **hash_what)`